submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,338 @@
+}
+
+function getResponseErrorMessage(status, statusText) {
+	return `<h1 class='error__message'>Sorry, an error occured!</h1>
+			<h2 class='error__code'>${status}: ${statusText}</h2>`;
+}
+
+function setAgeLimit(ageLimit, limitInputId, limitType) {
+	const limitInput = document.getElementById(limitInputId);
+	limitInput[limitType] = ageLimit;
+	setInputLimitValue(
+		limitInput,
+		limitInputId,
+		limitType,
+		"minAge",
+		"min",
+		ageLimit
+	);
+	setInputLimitValue(
+		limitInput,
+		limitInputId,
+		limitType,
+		"maxAge",
+		"max",
+		ageLimit
+	);
+}
+
+function setInputLimitValue(
+	limitInput,
+	limitInputId,
+	limitType,
+	limitInputIdValue,
+	limitTypeValue,
+	ageLimit
+) {
+	if (limitInputId === limitInputIdValue && limitType === limitTypeValue)
+		limitInput.value = ageLimit;
+}
+
+function getCertainAgeLimit(friendsArray, value) {
+	const sortedArray = friendsArray.sort((a, b) => a.age - b.age);
+	return value === "min"
+		? sortedArray[0].age
+		: sortedArray[sortedArray.length - 1].age;
+}
+
+function sortCards(e, friendsArray) {
+	if (e.target.classList.contains("list__item")) {
+		document.querySelector(".select__face-item").innerText =
+			e.target.textContent;
+		sortCardsArray(e.target.attributes.value.value, friendsArray);
+	}
+}
+
+function sortCardsArray(condition, friendsArray) {
+	switch (condition) {
+		case "ND":
+			FRIENDS_ARRAY.sort((a, b) => b.firstName.localeCompare(a.firstName));
+			break;
+		case "NA":
+			FRIENDS_ARRAY.sort((a, b) => a.firstName.localeCompare(b.firstName));
+			break;
+		case "AD":
+			FRIENDS_ARRAY.sort((a, b) => b.age - a.age);
+			break;
+		case "AA":
+			FRIENDS_ARRAY.sort((a, b) => a.age - b.age);
+			break;
+	}
+}
+
+function findSubstring(string, substring) {
+	return string.toLowerCase().indexOf(substring.toLowerCase()) >= 0
+		? true
+		: false;
+}
+
+function findMatchesWithPropertiesValues(
+	propertiesList,
+	friendsArray,
+	substring
+) {
+	return friendsArray.filter((friend) => {
+		return propertiesList
+			.map((property) => findSubstring(friend[property], substring))
+			.some((el) => el);
+	});
+}
+
+function filterByGenderValue(genderList, friendsArray) {
+	return friendsArray.filter((friend) =>
+		genderList.some((gender) => friend.gender === gender.value)
+	);
+}
+
+function filterByAgeLimits(minAge, maxAge, friendsArray) {
+	return friendsArray.filter(
+		(friend) => friend.age >= minAge && friend.age <= maxAge
+	);
+}
+
+function filterByGender(e, friendsArray) {
+	let checkboxes = document.querySelectorAll("input[type=checkbox]");
+	checkboxes = Array.from(checkboxes).filter((checkbox) => checkbox.checked);
+	return filterByGenderValue(checkboxes, friendsArray);
+}
+
+function filterByAge(friendsArray) {
+	const min = document.getElementById("minAge"),
+		max = document.getElementById("maxAge");
+	if (min.value && max.value) {
+		return filterByAgeLimits(min.value, max.value, friendsArray);
+	}
+}
+
+checkFiltersChanged = (event) =>
```

Comment:
Global variable? What for? Don't do this :)

Reply:
The coincidence :ok_woman: Fixed.